src/napari_spatialdata/_scatterwidgets.py

```diff
@@ -23,6 +30,89 @@
 ]
 
 
+class SelectFromCollection:
+    """
+    This class was taken from:
+    https://matplotlib.org/stable/gallery/widgets/lasso_selector_demo_sgskip.html
+
+    Select indices from a matplotlib collection using `LassoSelector`.
+
+    Selected indices are saved in the `ind` attribute. This tool fades out the
+    points that are not part of the selection (i.e., reduces their alpha
+    values). If your collection has alpha < 1, this tool will permanently
+    alter the alpha values.
+
+    Note that this tool selects collection objects based on their *origins*
+    (i.e., `offsets`).
+
+    Parameters
+    ----------
+    ax
+        Axes to interact with.
+    collection
+        Collection you want to select from.
+    alpha_other
+        To highlight a selection, this tool sets all selected points to an
+        alpha value of 1 and non-selected points to *alpha_other*.
+    """
+
+    def __init__(
+        self,
+        viewer: Viewer,
+        model: ImageModel,
+        ax: Axes,
+        collection: Collection,
+        data: List[NDArrayA],
+        alpha_other: float = 0.3,
+    ):
+        self.viewer = viewer
+        self.model = model
+        self.canvas = ax.figure.canvas
+        self.collection = collection
+        self.alpha_other = alpha_other
+        self.exported_data = None
+        self.data = data
+        self.axes = ax
+
+        self.xys = collection.get_offsets()
+        self.Npts = len(self.xys)
+
+        # Ensure that we have separate colors for each object
+        self.fc = collection.get_facecolors()
+
+        if len(self.fc) == 0:
+            raise ValueError("Collection must have a facecolor")
+        elif len(self.fc) == 1:
+            self.fc = np.tile(self.fc, (self.Npts, 1))
+
+        self.selector = LassoSelector(ax, onselect=self.onselect)
+
+        self.ind: Optional[NDArrayA] = None
+
+    def export(self, adata: AnnData) -> None:
+
+        model_layer: Layer = self.model.layer
+        obs_name = model_layer.name + "_LASSO_SELECTED"
```

Comment:
@giovp is there a better way to handle this? I got a flake8 error when not defining the type:
'error: Item "None" of "Optional[Any]" has no attribute "name"'